dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -0,0 +1,364 @@
+package com.dotmarketing.util;
+
+import com.dotcms.repackage.org.apache.commons.beanutils.PropertyUtils;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.links.model.Link;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.google.common.base.CaseFormat;
+import java.lang.reflect.Constructor;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class used to map query results to POJO objects
+ * @author nollymar
+ */
+public class ConvertToPOJOUtil {
```

Comment:
In general the idea is good, but I think we should use the Transformer pattern that we have already in place, such as 

Map<Class<T>, Transformer<T>> so that you can register later on more transformers or override and a default one for the rest of the case, we can talk about it